common/djangoapps/contentserver/middleware.py

```diff
@@ -55,6 +56,25 @@ def process_request(self, request):
             except (ItemNotFoundError, NotFoundError):
                 return HttpResponseNotFound()
 
+            # Set the basics for this request. Make sure that the course key for this
+            # asset has a run, which old-style courses do not.  Otherwise, this will
+            # explode when the key is serialized to be sent to NR.
+            safe_course_key = loc.course_key
+            if safe_course_key.run is None:
+                safe_course_key = safe_course_key.replace(run='only')
```

Comment:
Just struck me, this could probably be an empty string but then the `course_id` would come through as something like `edx/TESTING101/`, with that trailing slash at the end.  Any preference?  Right now, it'll be `edx/TESTING101/only`.